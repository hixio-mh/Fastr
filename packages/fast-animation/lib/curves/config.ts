import { IBezierCurve } from "./index";

export const linear: IBezierCurve = [
    [0, 0],
    [1, 1]
];
export const easeOut: IBezierCurve = [
    [0, 0],
    [0.58, 1]
];
export const easeOutSmooth: IBezierCurve = [
    [0, 0.35],
    [0.15, 1]
];
export const easeIn: IBezierCurve = [
    [0.25, 0.1],
    [0.25, 1]
];
export const drillIn: IBezierCurve = [
    [0.17, 0.17],
    [0, 1]
];
export const backToApp: IBezierCurve = [
    [0.5, 0],
    [0.6, 1]
];
export const appToApp: IBezierCurve = [
    [0.5, 0],
    [1, 0.9]
];
export const fastIn: IBezierCurve = [
    [0.1, 0.9],
    [0.2, 1]
];
export const fastOut: IBezierCurve = [
    [0.9, 0.1],
    [1, 0.2]
];
export const fastInOut: IBezierCurve = [
    [0.8, 0],
    [0.2, 1]
];
export const exponential: IBezierCurve = [
    [0.1, 0.25],
    [0.75, 0.9]
];
export const fastInFortySevenPercent: IBezierCurve = [
    [0.11, 0.5],
    [0.24, 0.96]
];
export const exponentialReversed: IBezierCurve = [
    [0.25, 0.1],
    [0.9, 0.75]
];
export const navPane: IBezierCurve = [
    [0.1, 0.7],
    [0.1, 1]
];
