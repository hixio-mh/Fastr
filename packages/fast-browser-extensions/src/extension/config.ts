/**
 * Define configuration values
 */
const validDomains: string[] = ["*://*.fluentweb.com/*"];
const extensionId: string = "jhhigejkemaekdcempcigaapebobmimg";

/**
 * Define event names
 */
export const menuItemClickEvent: string = "FW_EXTENSION::MENU_ITEM_CLICK";

/**
 * Non-production environments
 */
if (process.env.NODE_ENV !== "production") {
    validDomains.push("<all_urls>");
}

/**
 * Assign exports
 */
export { validDomains, extensionId };
