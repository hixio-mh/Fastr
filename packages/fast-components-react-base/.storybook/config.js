import React from "react";
import { configure, addParameters, addDecorator } from "@storybook/react";
import { create, Global } from "@storybook/theming";
// Import from dist cause storybook throws up if we import the named import from the index
import { theme } from "@microsoft/fast-storybook-presets";

addParameters({
    options: {
        theme,
    },
});
