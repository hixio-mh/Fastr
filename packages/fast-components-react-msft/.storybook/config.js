import React from "react";
import { configure, addParameters, addDecorator } from "@storybook/react";
import { create, Global } from "@storybook/theming";
import { addReadme } from "storybook-readme";

// Import from dist cause storybook throws up if we import the named import from the index
import theme from "@microsoft/fast-storybook-presets/dist/theme";

addDecorator(addReadme);
addParameters({
    options: {
        theme,
    },
});
