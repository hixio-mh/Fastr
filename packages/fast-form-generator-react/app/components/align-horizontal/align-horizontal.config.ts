import {
    IFormComponentMappingToPropertyNamesProps
} from "../../../src/form/form.props";

export default {
    alignHorizontal: [
        "alignHorizontal"
    ]
} as IFormComponentMappingToPropertyNamesProps;
