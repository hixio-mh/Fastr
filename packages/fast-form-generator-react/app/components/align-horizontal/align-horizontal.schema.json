{
    "$schema": "http://json-schema.org/schema#",
    "title": "Component with align horizontal",
    "description": "A test component's schema definition.",
    "type": "object",
    "id": "alignHorizontal",
    "properties": {
        "alignHorizontal": {
            "title": "Align horizontal",
            "type": "string",
            "enum": [
                "left",
                "center",
                "right"
            ]
        }
    },
    "required": [
        "alignHorizontal"
    ]
}
