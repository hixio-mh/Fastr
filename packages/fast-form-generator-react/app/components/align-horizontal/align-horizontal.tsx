import * as React from "react";

export enum AlignHorizontalPositions {
    left = "left",
    right = "right",
}

export interface AlignHorizontalProps {
    alignHorizontal: AlignHorizontalPositions;
}

/**
 * This test components API should have:
 * - a required property which maps to a configuration
 */
export default class AlignHorizontal extends React.Component<AlignHorizontalProps, {}> {
    public render(): JSX.Element {
        return <span>{this.props.alignHorizontal}</span>;
    }
}
