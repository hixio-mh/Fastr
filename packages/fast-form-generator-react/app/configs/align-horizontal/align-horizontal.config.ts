import { FormComponentMappingToPropertyNamesProps } from "../../../src/form/form.props";

export default {
    alignHorizontal: ["alignHorizontal", "optionalAlignHorizontal"],
} as FormComponentMappingToPropertyNamesProps;
