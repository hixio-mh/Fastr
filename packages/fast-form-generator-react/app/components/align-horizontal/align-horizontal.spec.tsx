import * as React from "react";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { SnapshotTestSuite } from "@microsoft/fast-jest-snapshots-react";
import { getExample } from "@microsoft/fast-permutator";
import Form, { FormProps } from "../../../src/form";
import config from "./align-horizontal.config";
import { AlignHorizontalProps } from "./align-horizontal";
import * as alignHorizontalSchema from "./align-horizontal.schema.json";

const name: string = "align-horizontal";

const exampleData: FormProps = {
    schema: alignHorizontalSchema,
    data: getExample(alignHorizontalSchema),
    /* tslint:disable-next-line */
    onChange: (data: any): void => {},
    componentMappingToPropertyNames: config,
};

const examples: SnapshotTestSuite<FormProps> = {
    name,
    component: Form,
    data: [exampleData],
};

describe(name, () => {
    generateSnapshots(examples);
});
