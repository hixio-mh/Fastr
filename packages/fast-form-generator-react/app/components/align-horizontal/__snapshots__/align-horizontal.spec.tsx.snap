// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align-horizontal align-horizontal: 0 1`] = `
<div
  className={null}
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <div>
          <div>
            <div
              className="formItemAlignHorizontal-0-1-11"
            >
              <label
                className="formItemAlignHorizontal_label-0-1-12"
                htmlFor="alignHorizontal"
              >
                Align horizontal
              </label>
              <div
                className="formItemAlignHorizontal_inputContainer-0-1-13"
              >
                <span>
                  <input
                    aria-label="left align"
                    checked={true}
                    className="formItemAlignHorizontal_input__left-0-1-14"
                    id="alignHorizontal"
                    name="alignHorizontal"
                    onChange={[Function]}
                    type="radio"
                    value="left"
                  />
                </span>
                <span>
                  <input
                    aria-label="center align"
                    checked={false}
                    className="formItemAlignHorizontal_input__center-0-1-15"
                    id="alignHorizontal"
                    name="alignHorizontal"
                    onChange={[Function]}
                    type="radio"
                    value="center"
                  />
                </span>
                <span>
                  <input
                    aria-label="right align"
                    checked={false}
                    className="formItemAlignHorizontal_input__right-0-1-16"
                    id="alignHorizontal"
                    name="alignHorizontal"
                    onChange={[Function]}
                    type="radio"
                    value="right"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
